drivers/nrf_802154_sl/src/platform/lp_timer/nrf_802154_lp_timer.c

```diff
@@ -0,0 +1,606 @@
+	nrf_rtc_int_disable(NRF_802154_RTC_INSTANCE,
+			    cmp_ch[LP_TIMER_CHANNEL].int_mask);
+	nrf_rtc_event_clear(NRF_802154_RTC_INSTANCE,
+			    cmp_ch[LP_TIMER_CHANNEL].event);
+}
+
+void nrf_802154_lp_timer_sync_start_now(void)
+{
+	uint32_t counter;
+	uint32_t offset;
+	uint64_t now;
+	uint32_t iterations = MAX_LP_TIMER_SYNC_ITERS;
+
+	do {
+		offset_and_counter_get(&offset, &counter);
+		now = time_get(offset, counter);
+		timer_sync_start_at((uint32_t)now, MIN_RTC_COMPARE_EVENT_DT,
+				    &now);
+	} while ((counter_get() != counter) && (--iterations > 0));
+}
+
+void nrf_802154_lp_timer_sync_start_at(uint32_t t0, uint32_t dt)
+{
+	uint64_t now = curr_time_get();
+
+	timer_sync_start_at(t0, dt, &now);
+}
+
+void nrf_802154_lp_timer_sync_stop(void)
+{
+	nrf_rtc_event_disable(NRF_802154_RTC_INSTANCE,
+			      cmp_ch[SYNC_CHANNEL].event_mask);
+	nrf_rtc_int_disable(NRF_802154_RTC_INSTANCE,
+			    cmp_ch[SYNC_CHANNEL].int_mask);
+	nrf_rtc_event_clear(NRF_802154_RTC_INSTANCE,
+			    cmp_ch[SYNC_CHANNEL].event);
+}
+
+uint32_t nrf_802154_lp_timer_sync_event_get(void)
+{
+	return nrf_rtc_event_address_get(NRF_802154_RTC_INSTANCE,
+					 cmp_ch[SYNC_CHANNEL].event);
+}
+
+uint32_t nrf_802154_lp_timer_sync_time_get(void)
+{
+	return (uint32_t)target_times[SYNC_CHANNEL];
+}
+
+void nrf_802154_clock_lfclk_ready(void)
+{
+	clock_ready = true;
+}
+
+static void rtc_irq_handler(void)
+{
+	/* Handle overflow. */
+	if (nrf_rtc_event_check(NRF_802154_RTC_INSTANCE,
+				NRF_RTC_EVENT_OVERFLOW)) {
+		/* Disable OVERFLOW interrupt to prevent lock-up in interrupt context while mutex is locked from lower priority context
+		 * and OVERFLOW event flag is stil up.
+		 * OVERFLOW interrupt will be re-enabled when mutex is released - either from this handler, or from lower priority context,
+		 * that locked the mutex. */
+		nrf_rtc_int_disable(NRF_802154_RTC_INSTANCE,
+				    NRF_RTC_INT_OVERFLOW_MASK);
+
+		/* Handle OVERFLOW event by reading current value of overflow counter. */
+		(void)overflow_counter_get();
+	}
+
+	/* Handle compare match. */
+	if (shall_fire_immediately) {
+		shall_fire_immediately = false;
+		handle_compare_match(true);
+	}
+
+	if (nrf_rtc_int_enable_check(NRF_802154_RTC_INSTANCE,
+				     cmp_ch[LP_TIMER_CHANNEL].int_mask) &&
+	    nrf_rtc_event_check(NRF_802154_RTC_INSTANCE,
+				cmp_ch[LP_TIMER_CHANNEL].event)) {
+		handle_compare_match(false);
+	}
+
+	if (nrf_rtc_int_enable_check(NRF_802154_RTC_INSTANCE,
+				     cmp_ch[SYNC_CHANNEL].int_mask) &&
+	    nrf_rtc_event_check(NRF_802154_RTC_INSTANCE,
+				cmp_ch[SYNC_CHANNEL].event)) {
+		nrf_rtc_event_clear(NRF_802154_RTC_INSTANCE,
+				    cmp_ch[SYNC_CHANNEL].event);
+		nrf_rtc_event_disable(NRF_802154_RTC_INSTANCE,
+				      cmp_ch[SYNC_CHANNEL].event_mask);
+		nrf_rtc_int_disable(NRF_802154_RTC_INSTANCE,
+				    cmp_ch[SYNC_CHANNEL].int_mask);
+		nrf_802154_lp_timer_synchronized();
+	}
+}
+
+void NRF_802154_RTC_IRQ_HANDLER(void)
+{
+	rtc_irq_handler();
+}
```

Comment:
This does not seem to be needed.

Reply:
Please note that the `lp_timer` module does not have several platform-dependent implementations and instead is supposed to be compliant with different platforms, e.g. Zephyr and bare-metal. Such a solution makes it possible to have just a single function (`rtc_irq_handler`) that can either be bound to the interrupt when compiled for Zephyr integration *or* just do the heavy-lifting in bare-metal variant. So while `NRF_802154_RTC_IRQ_HANDLER` is unneeded in RTOS-context, it makes it possible to have just a single implementation file.

Reply:
Right, I missed that it is not Zephyr specific. Sorry for the noise.